app/attributes/attributecontroller.cpp

```diff
@@ -249,8 +263,10 @@ void AttributeController::flatten(
               groupName,
               parentTabRow,
               layer->attributeDisplayName( fieldIndex ),
+              nameExpression,
               editorField->showLabel(),
               !isReadOnly,
+              editableExpression,
```

Comment:
Issue resolved: 
(Test2 with default value on update enabled is not editable even when `Editable` expression returns true)
 <img src="https://github.com/user-attachments/assets/8fb06fab-1bcc-4bbd-85a0-b3d4326b57f2" width="300">

Reply:
Cool, this is what I was referring to. However, the situation when you have both the editable expression and "apply default value on update" set, is _permanent_ - meaning that it will always be not editable and user can not do anything about it (ofc unless they switch off the `apply default value on update` in QGIS). If we add it there (https://github.com/MerginMaps/mobile/pull/3645/files#diff-1cac530c00b6b3114d6f3da9d8dc0fefd2a486a7eefc58814e9e8dd1c0da69acR931) we need to evaluate it every single time anything in the form changes. I was thinking we can do it in the `flatten` method where we call it just once. This way, if we detect that there is this `apply default value on update`, we do not even need to read the editable expression and simply keep the field read-only.


Cool, this is what I was referring to. However, when both the editable expression and `Apply default value on update` are set, the situation is _permanent_. In this state, the field will always be non-editable, and the user cannot change it (unless they disable the `Apply default value on update` option in QGIS).

If we include the fix here (I mean [here](https://github.com/MerginMaps/mobile/pull/3645/files#diff-1cac530c00b6b3114d6f3da9d8dc0fefd2a486a7eefc58814e9e8dd1c0da69acR931)), we would need to evaluate it every single time there is a change in the form (after each typed letter). Instead, I was thinking we could handle this in the flatten method, where it’s only called once.

This way, if we detect that `Apply default value on update` is enabled, we can bypass reading the editable expression entirely and simply keep the field read-only.